lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,180 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+sinatra_vs = Gem::Version.new(Sinatra::VERSION)
+sinatra_min_vs = Gem::Version.new('1.4.0')
+if sinatra_vs < sinatra_min_vs
+  raise "sinatra version #{sinatra_vs} is not supported yet " \
```

Comment:
It's because it's always better not to crash users apps just because of a well known incompatibility.